packages/framework/tests/Unit/Model/Order/Processing/OrderProcessorMiddleware/SetDeliveryAddressByDeliveryAddressUuidMiddlewareTest.php

```diff
@@ -0,0 +1,141 @@
+        $this->assertSame($expectedCountry, $actualOrderData->deliveryCountry);
+    }
+
+    /**
+     * @dataProvider noDeliveryIsSetDataProvider
+     * @param string|null $deliveryAddressUuid
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser|null $customerUser
+     * @param \Shopsys\FrameworkBundle\Model\Customer\DeliveryAddress|null $deliveryAddress
+     */
+    public function testNoDeliveryAddressIsSet(
+        ?string $deliveryAddressUuid,
+        ?CustomerUser $customerUser,
+        ?DeliveryAddress $deliveryAddress,
+    ): void {
+        $orderProcessingData = $this->createOrderProcessingData();
+
+        $orderProcessingData->orderInput->addAdditionalData(SetDeliveryAddressByDeliveryAddressUuidMiddleware::DELIVERY_ADDRESS_UUID, $deliveryAddressUuid);
+        $orderProcessingData->orderInput->setCustomerUser($customerUser);
+
+        $createSetDeliveryAddressByDeliveryAddressUuidMiddleware = $this->createSetDeliveryAddressByDeliveryAddressUuidMiddleware($deliveryAddress);
+
+        $result = $createSetDeliveryAddressByDeliveryAddressUuidMiddleware->handle($orderProcessingData, $this->createOrderProcessingStack());
+        $actualOrderData = $result->orderData;
+
+        $this->assertNull($actualOrderData->deliveryFirstName);
+        $this->assertNull($actualOrderData->deliveryLastName);
+        $this->assertNull($actualOrderData->deliveryCompanyName);
+        $this->assertNull($actualOrderData->deliveryTelephone);
+        $this->assertNull($actualOrderData->deliveryStreet);
+        $this->assertNull($actualOrderData->deliveryCity);
+        $this->assertNull($actualOrderData->deliveryPostcode);
+        $this->assertNull($actualOrderData->deliveryCountry);
+    }
+
+    /**
+     * @return iterable
+     */
+    public function noDeliveryIsSetDataProvider(): iterable
+    {
+        yield 'no delivery address uuid' => [
+            null,
+            $this->createCustomerUser(),
+            $this->createMock(DeliveryAddress::class),
+        ];
+
+        yield 'no customer user' => [
+            '30a07736-85aa-4407-8a85-4e722ed3e035',
+            null,
+            $this->createMock(DeliveryAddress::class),
+        ];
+
+        yield 'no delivery address' => [
+            '30a07736-85aa-4407-8a85-4e722ed3e035',
+            $this->createCustomerUser(),
+            null,
+        ];
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\DeliveryAddress|null $deliveryAddress
+     * @return \Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessorMiddleware\SetDeliveryAddressByDeliveryAddressUuidMiddleware
+     */
+    private function createSetDeliveryAddressByDeliveryAddressUuidMiddleware(
+        ?DeliveryAddress $deliveryAddress,
+    ): SetDeliveryAddressByDeliveryAddressUuidMiddleware {
+        $deliveryAddressFacadeMock = $this->createMock(DeliveryAddressFacade::class);
+
+        $deliveryAddressFacadeMock
+            ->method('findByUuidAndCustomer')
```

Comment:
would it make sense to add an expectation for an input argument?
```
->with($deliveryAddressUuid)
```
And pass `$deliveryAddressUuid` here as an argument for `createSetDeliveryAddressByDeliveryAddressUuidMiddleware()` :thinking:

Reply:
is it necessary?

Reply:
for me, it seemed like a logical link for better understanding the test, otherwise, calling `$orderProcessingData->orderInput->addAdditionalData(SetDeliveryAddressByDeliveryAddressUuidMiddleware::DELIVERY_ADDRESS_UUID, $deliveryAddressUuid);` seems pointless, is not it?

Reply:
refactored a little bit